docs/2019-09-09-Combining-Rewrite-Axioms.md

```diff
@@ -0,0 +1,172 @@
+
+
+In the
+[Polkadot Verification](https://github.com/runtimeverification/polkadot-verification/issues/20)
+issue, @ehildenb proposed that we should either apply a chain of rules to a
+symbolic configuration consisting of a single variable, or we should be able to
+combine rules into a single one. This document details the matching logic
+reasoning and assumptions behind these.
+
+Summary
+=======
+
+If we have this chain of axioms: `α₁(X₁)⇒β₁(X₁)` … `αₙ(Xₙ)⇒βₙ(Xₙ)` in which
+all the LHS are *functional* and *injection-based* then their combined
+transition is
```

Comment:
I thought maybe we needed the LHS to be _constructors_, but this seems like a weaker condition? Is it still enough?

Reply:
If I didn't make any mistake, the only place where we use the constructor properties is when solving `α(Y)=α(Y')` to get `Y'=Y`, and, for that, we only need the injectivity property: we have exactly the same symbols in both terms, so we don't need, say, the no-confusion-different-constructors property, or the no-junk one.